.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+# Contributing
+ 
+When contributing to this repository, please first discuss the change you wish to make via issue,
+or [Hypertrace slack](https://bit.ly/hypertrace-slack) with the owners of this repository before making a change.
+ 
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+ 
+## Submitting bugs
+ 
+### Due diligence
+-------------
+ 
+Before submitting a bug, please do the following:
+ 
+* Perform **basic troubleshooting** steps:
+ 
+   * **Make sure you're on the latest version.** If you're not on the most
+     recent version, your problem may have been solved already! Upgrading is
+     always the best first step.
+   * **Try older versions.** If you're already *on* the latest release, try
+     rolling back a few minor versions (e.g. if on 1.7, try 1.5 or 1.6) and
+     see if the problem goes away. This will help the devs narrow down when
+     the problem first arose in the commit log.
+   * **Try switching up dependency versions.** If the software in question has
+     dependencies (other libraries, etc) try upgrading/downgrading those as
+     well.
+ 
+* **Search the GitHub issues** to make sure it's not a known
+ issue.
+* If you don't find a pre-existing issue, consider **checking with the slack channel** in case the problem is non-bug-related.
+ 
+### What to put in your issue
+------------------------------
+ 
+Make sure issue you raised gets the attention it deserves: issues with missing
+information may be ignored or punted back to you, delaying a fix.  The below
+constitutes a bare minimum; more info is almost always better:
+ 
+* **What version of the core programming language interpreter/compiler are you
+ using?** For example, Most of our services are written in Java so you what version of Java you're on?
+* **What operating system are you on?** Windows? (10? Home? Pro?)
+ Mac OS X?  (10.x.x?) Linux? (Which distro? Which version of that
+ distro? 32 or 64 bits?) Again, more detail is better.
+* **Are you using latest codebase and are on main branch?** Ideally, you
+ followed the advice above and have ruled out (or verified that the problem
+ exists in) a few different versions.
+* **How can the developers recreate the bug on their end?** If possible,
```

Comment:
If possible, include a copy of your code *or an example repo*, the command you used to invoke it, and the full
 output of your run (if applicable.)

Reply:
should we completely remove this section and add this things in issue template?

Reply:
done!